File: 234-Palindrome-Linked-List/py/main.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def is_palindrome(head: ListNode) -> bool:
    head_list = []

    while head:
        head_list.append(head.val)
        head = head.next

    first = 0
    length = len(head_list)
    last = length - 1

    i = 0
    loop_end = length // 2
    while i < loop_end:
        if head_list[first] != head_list[last]:
            return False
        first += 1
        last -= 1
        i += 1

    return True

File: 234-Palindrome-Linked-List/ts/main.ts
class ListNode {
    val: number;
    next: ListNode | null;
    
    constructor(val: number = 0, next: ListNode | null = null) {
        this.val = val;
        this.next = next;
    }
}

function isPalindrome(head: ListNode | null): boolean {
    let head_vec: number[] = [];
    
    while (head !== null) {
        head_vec.push(head.val);
        head = head.next;
    }
    
    let first = 0;
    let length = head_vec.length;
    let last = length - 1;
    
    let i = 0;
    let loop_end = Math.floor(length / 2);
    while (i < loop_end) {
        if (head_vec[first] !== head_vec[last]) {
            return false;
        }
        first++;
        last--;
        i++;
    }
    
    return true;
}
